CMakeLists.txt

```diff
@@ -1,10 +1,29 @@
 CMAKE_MINIMUM_REQUIRED(VERSION 3.2 FATAL_ERROR)
 PROJECT(tflite_micro_compiler)
+SET(CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake)
 
-SET(TF_DIR "../tensorflow" CACHE STRING "TensorFlow source directory")
-SET(TFL_SRC ${TF_DIR}/tensorflow/lite)
-SET(TFLM_SRC ${TFL_SRC}/micro)
-SET(TFLMD_SRC ${TFLM_SRC}/tools/make/downloads)
+IF(NOT TF_DIR)
+    SET(TF_DIR "../tensorflow" CACHE STRING "TensorFlow source directory")
+ENDIF()
+
+GET_FILENAME_COMPONENT(tf_fullpath ${TF_DIR} REALPATH)
+IF(EXISTS "${tf_fullpath}" AND NOT GET_TF_SRC)
```

Comment:
We could be more explicit and do this decision just on GET_TF_SRC. Then move the path check down and give a hard error if it doesn't exist.